src/PKSim.Core/Snapshots/Mappers/EventMapper.cs

```diff
@@ -16,10 +15,10 @@ public EventMapper(ParameterMapper parameterMapper, IEventFactory eventFactory)
 
       public override SnapshotEvent MapToSnapshot(ModelEvent modelEvent)
       {
-         var snapshotEvent = new SnapshotEvent {Template = modelEvent.TemplateName};
-         MapModelPropertiesIntoSnapshot(modelEvent, snapshotEvent);
-         MapVisibleParameters(modelEvent, snapshotEvent);
-         return snapshotEvent;
+         return SnapshotFrom(modelEvent, snapshot =>
```

Comment:
The common method is starting to shape up nicely. The actual code in each container becomes trivial